log/testutil.go

```diff
@@ -0,0 +1,122 @@
+package log
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+// FieldValidator facilitates testing of Span.Log*() implementations.
+//
+// FieldValidator SHOULD ONLY BE USED FOR TESTS. It is not intended for
+// production code.
+//
+// Usage:
+//
+//     fv := log.NewFieldValidator(t, someLogStructure.Fields)
+//     fv.
+//         ExpectNextFieldEquals("key1", reflect.String, "some string value").
+//         ExpectNextFieldEquals("key2", reflect.Uint32, "4294967295")
```

Comment:
not sure I see a strong need for this in the API package. E.g. you didn't need to use it in mock tracer tests. Whatever the internal format of field storage a tracer implementation is using, they should be able to do assertions directly against that struct, like https://github.com/opentracing/opentracing-go/pull/108/files#diff-6d19a913e5779231470a18e52e75389fR180